packages/health/lib/src/health_factory.dart

```diff
@@ -348,10 +403,19 @@ class HealthFactory {
     return stepsCount;
   }
 
+  /// Assigns numbers to specific [HealthDataType]s.
   int _alignValue(HealthDataType type) {
     switch (type) {
-      case HealthDataType.SLEEP:
+      case HealthDataType.SLEEP_IN_BED:
         return 0;
+      case HealthDataType.SLEEP_ASLEEP:
+        return 1;
+      case HealthDataType.SLEEP_AWAKE:
+        return 2;
+      case HealthDataType.SLEEP_DEEP:
+        return 3;
+      case HealthDataType.SLEEP_REM:
+        return 4;
       case HealthDataType.HEADACHE_UNSPECIFIED:
```

Comment:
are you sure it suppose to be same as sleep in bed

Reply:
In overall yes, as what this method does, it is maps the enum to numerical representation on the native side.